src/js_native_api_v8.cc

```diff
@@ -1662,6 +1662,28 @@ napi_status napi_create_symbol(napi_env env,
   return napi_clear_last_error(env);
 }
 
+napi_status napi_create_symbol_for(napi_env env,
+                                   napi_value description,
```

Comment:
Why not just `const char*`

Reply:
If we use const char* it would limit the description string to only latin1 or utf8. Accepting a js value (represented by napi_value) would allow description strings of types - utf8, utf16 and latin1, all at once.

Reply:
> If we use const char* it would limit the description string to only latin1 or utf8

Fwiw, that doesn’t really make sense, the description is limited to JS strings anyway. That means that using only UTF-8 would be just fine.

I don’t have strong feelings here, but there is definite precedent for this, e.g. in the form of `napi_set_named_property
()` which also takes a UTF-8 `const char*` for the same reason.

Reply:
Oh, I had incorrectly assumed that V8 preserves the encoding format of the strings.

Even if we change napi_create_symbol_for() to accept a const char*, it would be inconsistent with the sibling function, napi_create_symbol(), which still accepts a napi_value and that is unlikely to change as it would be ABI breaking.

Reply:
> I had incorrectly assumed that V8 preserves the encoding format of the strings.

Sometimes it does, sometimes it doesn’t (in particular, I don’t think V8 ever uses UTF-8 as an internal representation for strings). It doesn’t matter though, the behavior is not visible to users.